packages/shared-components/src/hooks/useAuditClient.ts

```diff
@@ -0,0 +1,113 @@
+import type { PublicAuditEvent } from '@klicker-uzh/types'
+import { useCallback, useState } from 'react'
+
+interface UseAuditClientOptions {
+  enabled?: boolean
+  assessmentMode?: boolean
+  baseUrl?: string
+  onError?: (error: Error) => void
+}
+
+interface AuditClientAPI {
+  log: (event: PublicAuditEvent) => Promise<void>
+  logAsync: (event: PublicAuditEvent) => void
+  isLoading: boolean
+  error: Error | null
+}
+
+/**
+ * Frontend audit client hook for logging user actions
+ * Only logs events when in assessment mode
+ * Includes simple retry mechanism with exponential backoff
+ */
+export default function useAuditClient(
+  options: UseAuditClientOptions = {}
+): AuditClientAPI {
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<Error | null>(null)
+
+  const {
+    enabled = process.env.NEXT_PUBLIC_AUDIT_ENABLED === 'true',
+    assessmentMode = false,
+    baseUrl = process.env.NEXT_PUBLIC_AUDIT_SERVICE_URL || '',
+    onError,
+  } = options
+
+  const log = useCallback(
+    async (event: PublicAuditEvent) => {
+      // Only log in assessment mode
+      if (!enabled || !assessmentMode) {
+        return
+      }
+
+      setIsLoading(true)
+      setError(null)
+
+      let lastError: Error | null = null
+      const maxRetries = 3
+
+      // Try up to 3 times with exponential backoff
+      for (let attempt = 0; attempt < maxRetries; attempt++) {
+        try {
+          const url = `${baseUrl}/audit/public`
+          const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            credentials: 'include', // Automatically sends domain cookies
+            body: JSON.stringify(event),
+          })
+
+          if (!response.ok) {
+            throw new Error(
+              `Audit logging failed: ${response.status} ${response.statusText}`
+            )
+          }
+
+          // Success - exit retry loop
+          setIsLoading(false)
+          return
+        } catch (err) {
+          lastError =
+            err instanceof Error ? err : new Error('Unknown audit error')
+
+          // Wait before retry with exponential backoff (500ms, 1s, 2s)
+          if (attempt < maxRetries - 1) {
+            const delay = 500 * Math.pow(2, attempt)
+            await new Promise((resolve) => setTimeout(resolve, delay))
+          }
+        }
+      }
+
+      // All retries failed
+      setError(lastError)
```

Comment:
we would need to decide what we do if even though it retried it still returns an error. probably let the user know with a toast? maybe its also just their internet that is down and the main requests fail anyways